modules/annotations/shared/src/test/scala/io/circe/derivation/annotations/JsonCodecMacrosSuite.scala

```diff
@@ -158,67 +159,67 @@ class JsonCodecMacrosSuite extends CirceSuite {
     Encoder[CaseClass3]
   }
 
-  it should "generate the correct JSON" in {
+  test("generate the correct JSON") {
     @JsonCodec(config = Configuration.default)
     case class CaseClass(fooCamel: String, barCamel: Int)
 
     val expectedJson = """{"fooCamel":"foo","barCamel":1}"""
     val generatedJson = Encoder[CaseClass].apply(CaseClass("foo", 1)).noSpaces
 
-    assertEq(expectedJson, generatedJson)
+    assertEquals(expectedJson, generatedJson)
   }
 
-  it should "generate snake case JSON" in {
+  test("generate snake case JSON") {
     @JsonCodec(config = Configuration.default.withSnakeCaseMemberNames)
     case class CaseClass(fooSnake: String, barSnake: Int)
 
     val generatedJson = Encoder[CaseClass].apply(CaseClass("foo", 1)).noSpaces
     val expectedJson = """{"foo_snake":"foo","bar_snake":1}"""
 
-    assertEq(expectedJson, generatedJson)
+    assertEquals(expectedJson, generatedJson)
   }
 
-  it should "generate kebab case JSON" in {
+  test("generate kebab case JSON") {
     @JsonCodec(config = Configuration.default.withKebabCaseMemberNames)
     case class CaseClass(fooKebab: String, barKebab: Int)
 
     val generatedJson = Encoder[CaseClass].apply(CaseClass("foo", 1)).noSpaces
     val expectedJson = """{"foo-kebab":"foo","bar-kebab":1}"""
 
-    assertEq(expectedJson, generatedJson)
+    assertEquals(expectedJson, generatedJson)
   }
 
-  "@JsonCodec(config = Configuration.decodeOnly)" should "provide Decoder instances" in {
+  test("@JsonCodec(config = Configuration.decodeOnly) should provide Decoder instances") {
     @JsonCodec(config = Configuration.decodeOnly)
     case class CaseClassDecodeOnly(foo: String, bar: Int)
 
     Decoder[CaseClassDecodeOnly]
-    assertDoesNotCompile("Encoder[CaseClassDecodeOnly]")
+    assert(compileErrors("Encoder[CaseClassDecodeOnly]").nonEmpty)
```

Comment:
Not sure is this is the best replacement for `assertDoesNotCompile`.
Would it make sense to add `clue()` around the compile errors :thinking: